From 88748df984fbccc3cce99af2c152d196b911e030 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:10:59 -0400
Subject: [powerpc] module: handle MODVERSION for .TOC

Message-id: <1410545655-205645-431-git-send-email-dzickus@redhat.com>
Patchwork-id: 94285
O-Subject: [RHEL7 PATCH 430/626] powerpc: module: handle MODVERSION for .TOC.
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 71ec7c55ed91e2352c00d51d171fccaa7cef5a00
Author: Rusty Russell <rusty@rustcorp.com.au>
Date:   Tue Mar 18 19:59:11 2014 +1030

    powerpc: module: handle MODVERSION for .TOC.

    For the ELFv2 ABI, powerpc introduces a magic symbol ".TOC.".  If we
    don't create a CRC for it (minus the leading ".", since we strip that)
    we get a modpost warning about missing CRC and the CRC array seems to
    be displaced by 1 so other CRCs mismatch too.

    Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/misc_64.S b/arch/powerpc/kernel/misc_64.S
index 8901489..4abfd29 100644
--- a/arch/powerpc/kernel/misc_64.S
+++ b/arch/powerpc/kernel/misc_64.S
@@ -603,6 +603,15 @@ _GLOBAL(kexec_sequence)
 
 #ifdef CONFIG_MODULES
 #if defined(_CALL_ELF) && _CALL_ELF == 2
+
+#ifdef CONFIG_MODVERSIONS
+.weak __crc_TOC.
+.section "___kcrctab+TOC.","a"
+.globl __kcrctab_TOC.
+__kcrctab_TOC.:
+ .llong __crc_TOC.
+#endif
+
 /*
  * Export a fake .TOC. since both modpost and depmod will complain otherwise.
  * Both modpost and depmod strip the leading . so we do the same here.
-- 
1.7.1